portfolio/src/main/webapp/script.js

```diff
@@ -53,30 +53,46 @@ function loadSkills () {
 
 /**
  * Makes a request to /data and gets the response
+ * if stay is true, then the buttons won't update
  */
-async function fetchFromData() {
+async function fetchFromData(stay) {
```

Comment:
It sounds to me you can move "hide/show msg button" logic to a separate function, so that you don't need to pass "stay" here. It is kind of unrelated to "fetching data".